### PR TITLE
update dynamodb definition

### LINKED ISSUE #132: Update DynamoDB definition
## Feature Title

Update DynamoDB definition

## User Stories

Since the specification has changed, we need to update the DynamoDB setting.

## Acceptance Criteria

- [x] Change GSI setting in the CDK
- [x] Change GSI setting in the main function

## Technical Details

- API Changes:
- Database Changes:
- Dependencies:

## Additional Context

None

### PR DESCRIPTION
**Pull request by hitohata**
## Overviews of implementation

Added a new GSI `EMailAndUserIdRelationship`

### CODE REVIEW

File: backend/main/others/dynamodbInitializer/src/index.ts [hunk 1]
```diff
@@ -90,6 +94,22 @@ const input: CreateTableCommandInput = {
 				WriteCapacityUnits: 1,
 			},
 		},
+		{
+			IndexName: "EMailAndUserIdRelationship",
+			KeySchema: [
+				{
+					AttributeName: "EMailAddress",
+					KeyType: "HASH",
+				},
```
Comment (taniguchiiqqq) [inline backend/main/others/dynamodbInitializer/src/index.ts:103]: Don't we need UserId as SortKey in this GSI?
```
{
	AttributeName: "UserId",
	KeyType: "RANGE",
},
```

**Review by taniguchiiqqq — COMMENTED** [review]
May I ask how I can use the added GSI? I try to figure it out by using NoSQL Workbench.

What I expected is that I can see the GSI on the operation builder view like this.
image

But when I init the Dynamo DB by the this code, the "EMailAndUserIdRelationship" GSI didn't show up like this.
image

File: backend/main/others/dynamodbInitializer/src/index.ts (hunk 1 above)
Comment (hitohata) [reply backend/main/others/dynamodbInitializer/src/index.ts:103]: I do not think so.
The email address must be unique on its own. Therefore, it doesn't need additional data.

**Comment by hitohata** [comment]
I don't know what has occurred. But in my environment, I can see three GSIs.

image

Please make sure, it is connected correctly. In this PR, the table name is Nishiki-DB, but your screenshot is `Nishiki-Table`.

https://github.com/genesis-tech-tribe/nishiki-backend/blob/feature/132-update-dynamodb-definition/backend/main/others/dynamodbInitializer/src/index.ts

Anyway, the connection to the workbench is not a concern of this PR. It's out of this PR scope.

File: backend/main/others/dynamodbInitializer/src/index.ts (hunk 1 above)
Comment (taniguchiiqqq) [reply backend/main/others/dynamodbInitializer/src/index.ts:103]: I'm so sorry. I misunderstood. Now I got why the UserId isn't needed. Thank you!

**Review by taniguchiiqqq — APPROVED** [review]
@hitohata 
> I don't know what has occurred. But in my environment, I can see three GSIs.
After I made sure the connection, I couldn't see three GSI. but you're right, it's beyond this PR's scope. so I'll approve it for now.

**Pull request merged**